dbms/src/Flash/Planner/plans/PhysicalJoin.cpp

```diff
@@ -78,11 +81,43 @@ void executeUnionForPreviousNonJoinedData(DAGPipeline & probe_pipeline, Context
 }
 } // namespace
 
+/// Return pair: first for the ExchangeReceiver node count; second for the total sum of source num
+std::pair<size_t, size_t> getReceiverSourceNumInfo(const PhysicalPlanNodePtr & node)
+{
+    switch (node->tp())
+    {
+    case PlanType::PlanTypeEnum::ExchangeReceiver:
+    {
+        const auto * receiver_ptr = dynamic_cast<const PhysicalExchangeReceiver *>(node.get());
+        RUNTIME_ASSERT(receiver_ptr);
+        return std::make_pair(1, receiver_ptr->getSourceNum());
+    }
+    case PlanType::PlanTypeEnum::MockExchangeReceiver:
+    {
+        const auto * mock_receiver_ptr = dynamic_cast<const PhysicalMockExchangeReceiver *>(node.get());
+        RUNTIME_ASSERT(mock_receiver_ptr);
+        return std::make_pair(1, mock_receiver_ptr->getSourceNum());
+    }
+    default:
```

Comment:
It seems that `PhysicalJoin`'s child must be `MockExchangeReceiver` or `ExchangeReceiver` here.

Reply:
throw Exception directly?

Reply:
Yeah, currently it is. While there could be projection in theory, that's why the code looked like this.

Reply:
Haha, not add limitations inside the method, just don't want to make the method too specific:)